src/main/java/hudson/plugins/jira/JiraFolderProperty.java

```diff
@@ -31,42 +27,30 @@
     public JiraFolderProperty() {
     }
 
-    @Override
-    public AbstractFolderProperty<?> reconfigure(StaplerRequest req, JSONObject formData)
-            throws Descriptor.FormException {
-        if (formData == null) {
-            return null;
-        }
-        //Fix^H^H^HDirty hack for empty string to URL conversion error
-        //Should check for existing handler etc, but since this is a dirty hack,
-        //we won't
-        Stapler.CONVERT_UTILS.deregister(java.net.URL.class);
-        Stapler.CONVERT_UTILS.register(new EmptyFriendlyURLConverter(), java.net.URL.class);
-        //End hack
-
-        sites.replaceBy(req.bindJSONToList(JiraSite.class, formData.get("sites")));
-        return this;
-    }
-
     /**
      * Return the JIRA sites.
      *
      * @return the JIRA sites
      */
-    public JiraSite[] getSites() {
```

Comment:
NO do not break backward compat they might be plugins using that.

Reply:
github is not the only place in the world where people write code :)

Reply:
They can always adapt when they update their version then 🙊

Reply:
I think we're discussing here basically which version this will be released as in terms of https://semver.org/. There are two sides here:
* one is the downstream backward compatibility (for plugins that inherit from this plugin).
* second is the upstream Jenkins compatibility (for the changes that were needed to adapt this plugin to Jenkins API changes)

For the open source contributions to work smoothly (where a downstream plugin might pick up the changes in half a year from the upstream), we should `@Deprecate` methods if possible to have a _transition period_ of adaptation.

Reply:
and when there is no transition? You cannot have two method with different return types.